src/templates/filter-bar-controller.js

```diff
@@ -29,12 +29,17 @@ FilterBarController.prototype.selectFilter = function(filterName) {
 };
 
 FilterBarController.prototype.getFilteredNotes = function(notes) {
-    const activeTab = Filters[getFilterName(store.getState())];
+    const filterName = getFilterName(store.getState());
+    const activeTab = Filters[filterName];
     if (!notes || !activeTab) {
         return [];
     }
 
-    return notes.filter(activeTab().filter);
+    const filteredNoteReads = store.getState().notes.filteredNoteReads;
+    const filterFunction = (note) =>
+        (filterName === 'unread' && filteredNoteReads.includes(note.id)) || activeTab().filter(note);
```

Comment:
note that we have no IE support for `Array.includes()` so we should either use `lodash.includes()` or change the data structure

```js
if ( SELECT_NOTE )
	return { ...state, [ note.id ]: true };

…

noteHasFilteredRead( note.id )

…

noteHasFilteredRead = ( state, noteId ) => !! getNotesState( state ).filteredNotes[ noteId ];
```